src/index.tsx

```diff
@@ -13,50 +27,67 @@ type ProvidedContext = {
   loading: boolean;
   prefab: typeof prefab;
   keys: string[];
+  settings: SharedSettings;
 };
 
-const defaultContext: ProvidedContext = {
+export const defaultContext: ProvidedContext = {
   get: (_: string) => undefined,
   getDuration: (_: string) => undefined,
   isEnabled: (_: string) => false,
   keys: [],
   loading: true,
   contextAttributes: {},
   prefab,
+  settings: {},
 };
 
 const PrefabContext = React.createContext(defaultContext);
 
 const usePrefab = () => React.useContext(PrefabContext);
 
-type EvaluationCallback = (key: string, value: ConfigValue, context: Context | undefined) => void;
+let globalPrefabIsTaken = false;
+
+const assignPrefabClient = () => {
+  if (globalPrefabIsTaken) {
+    return new Prefab();
+  }
 
-type Props = {
-  apiKey: string;
+  globalPrefabIsTaken = true;
+  return prefab;
+};
+
+type Props = SharedSettings & {
   contextAttributes?: ContextAttributes;
-  endpoints?: string[];
-  apiEndpoint?: string;
-  timeout?: number;
-  pollInterval?: number;
-  onError?: (error: Error) => void;
-  afterEvaluationCallback?: EvaluationCallback;
-  collectEvaluationSummaries?: boolean;
-  collectLoggerNames?: boolean;
 };
+const defaultOnError = () => {};
+const defaultCollectEvaluationSummaries = false;
+const defaultCollectLoggerNames = false;
```

Comment:
Did you have to pull these out for typing reasons? Also I thought you had changed the default on `collectEvaluationSummaries` to `true` recently

Reply:
I did, but they don't need to be anymore.

The latter here is a bug, let's fix it. I changed the default in the JS library but forgot we also had a default here 🤦